src/runtime/hashmap.go

```diff
@@ -154,34 +153,61 @@ func hashmapLenUnsafePointer(m unsafe.Pointer) int {
 	return hashmapLen((*hashmap)(m))
 }
 
+//go:inline
+func hashmapBucketSize(m *hashmap) uintptr {
+	return unsafe.Sizeof(hashmapBucket{}) + uintptr(m.keySize)*8 + uintptr(m.valueSize)*8
+}
+
+//go:inline
+func hashmapBucketAddr(m *hashmap, buckets unsafe.Pointer, n uintptr) *hashmapBucket {
+	bucketSize := hashmapBucketSize(m)
+	bucket := (*hashmapBucket)(unsafe.Add(buckets, bucketSize*n))
+	return bucket
+}
+
+//go:inline
+func hashmapBucketAddrForHash(m *hashmap, hash uint32) *hashmapBucket {
+	numBuckets := uintptr(1) << m.bucketBits
+	bucketNumber := (uintptr(hash) & (numBuckets - 1))
+	return hashmapBucketAddr(m, m.buckets, bucketNumber)
+}
+
+//go:inline
+func hashmapSlotKey(m *hashmap, bucket *hashmapBucket, slot uint8) unsafe.Pointer {
+	slotKeyOffset := unsafe.Sizeof(hashmapBucket{}) + uintptr(m.keySize)*uintptr(slot)
+	slotKey := unsafe.Pointer(uintptr(unsafe.Pointer(bucket)) + slotKeyOffset)
+	return slotKey
+}
+
+//go:inline
+func hashmapSlotValue(m *hashmap, bucket *hashmapBucket, slot uint8) unsafe.Pointer {
+	slotValueOffset := unsafe.Sizeof(hashmapBucket{}) + uintptr(m.keySize)*8 + uintptr(m.valueSize)*uintptr(slot)
+	slotValue := unsafe.Pointer(uintptr(unsafe.Pointer(bucket)) + slotValueOffset)
+	return slotValue
+}
+
```

Comment:
I appreciate you adding these helper functions, makes the code a lot more readable